testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/management/console/HttpManagementConstantHeadersTestCase.java

```diff
@@ -0,0 +1,328 @@
+      assertEquals(200, response.getStatusLine().getStatusCode());
+
+      Header header = response.getFirstHeader("X-All");
+      assertEquals("All", header.getValue());
+
+      header = response.getFirstHeader("X-Error");
+      assertEquals("Error", header.getValue());
+
+      header = response.getFirstHeader("X-Management");
+      assertNull("Header X-Management Unexpected", header);
+   }
+
+   /**
+    * Test that security headers can be configured and are returned from '/console' endpoint
+    */
+   @Test
+   public void testSecurity() throws Exception {
+      Map<String, List<Map<String, String>>> headersMap = new HashMap<>();
+
+      List<Map<String, String>> headers = new LinkedList<>();
+      headers.add(Collections.singletonMap("X-XSS-Protection", "1; mode=block"));
+      headers.add(Collections.singletonMap("X-Frame-Options", "SAMEORIGIN"));
+      headers.add(Collections.singletonMap("Content-Security-Policy", "default-src https: data: 'unsafe-inline' 'unsafe-eval'"));
+      headers.add(Collections.singletonMap("Strict-Transport-Security", "max-age=31536000; includeSubDomains;"));
+      headers.add(Collections.singletonMap("X-Content-Type-Options", "nosniff"));
+
+      headersMap.put(CONSOLE_CTX, headers);
+
+      managementClient.getControllerClient().execute(createConstantHeadersOperation(headersMap));
+      ServerReload.executeReloadAndWaitForCompletion(managementClient.getControllerClient());
+
+      HttpGet get = new HttpGet(managementConsoleUrl.toURI().toString());
+      HttpResponse response = httpClient.execute(get);
+      assertEquals(200, response.getStatusLine().getStatusCode());
+
+      Header header = response.getFirstHeader("X-XSS-Protection");
+      assertEquals("1; mode=block", header.getValue());
+
+      header = response.getFirstHeader("X-Frame-Options");
+      assertEquals("SAMEORIGIN", header.getValue());
+
+      header = response.getFirstHeader("Content-Security-Policy");
+      assertEquals("default-src https: data: 'unsafe-inline' 'unsafe-eval'", header.getValue());
+
+      header = response.getFirstHeader("Strict-Transport-Security");
+      assertEquals("max-age=31536000; includeSubDomains;", header.getValue());
+
+      header = response.getFirstHeader("X-Content-Type-Options");
+      assertEquals("nosniff", header.getValue());
+   }
+
+   /**
+    * Test that configured headers override original headers set by /console endpoint (except for disallowed headers)
+    */
+   @Test
+   public void testHeadersOverride() throws Exception {
+      HttpGet get = new HttpGet(managementConsoleUrl.toURI().toString());
+      HttpResponse response = httpClient.execute(get);
+      assertEquals(200, response.getStatusLine().getStatusCode());
+
+      Header[] headerArray = response.getAllHeaders();
+
+      List<Header> headerList = Arrays.stream(headerArray)
+            .filter(header -> !header.getName().equals("Connection") && !header.getName().equals("Date")
```

Comment:
Probably not the end of the world as this is in a test but FYI streaming an array that could have been iterated in a for loop is very expensive, a lot of intermediate objects needs to be created and garbage collected to do something that could have been performed with minimal objects.

Don't change the PR for this as I am looking to merge but more a FYI.